code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -204,78 +204,103 @@
+		mecha_flags ^= ID_LOCK_ON
+		to_chat(user, span_warning("ID lock [mecha_flags & ID_LOCK_ON ? "enabled" : "disabled"]."))
 		return
 
-	if(istype(W, /obj/item/stock_parts/cell))
-		if(construction_state == MECHA_OPEN_HATCH)
-			if(!cell)
-				if(!user.transferItemToLoc(W, src, silent = FALSE))
-					return
-				var/obj/item/stock_parts/cell/C = W
-				to_chat(user, span_notice("You install the power cell."))
-				playsound(src, 'sound/items/screwdriver2.ogg', 50, FALSE)
-				cell = C
-				log_message("Power cell installed", LOG_MECHA)
-			else
-				to_chat(user, span_warning("There's already a power cell installed!"))
+	if(istype(weapon, /obj/item/mecha_parts))
+		var/obj/item/mecha_parts/part = weapon
+		part.try_attach_part(user, src, FALSE)
 		return
 
-	if(istype(W, /obj/item/stock_parts/scanning_module))
-		if(construction_state == MECHA_OPEN_HATCH)
-			if(!scanmod)
-				if(!user.transferItemToLoc(W, src, silent = FALSE))
-					return
-				to_chat(user, span_notice("You install the scanning module."))
-				playsound(src, 'sound/items/screwdriver2.ogg', 50, FALSE)
-				scanmod = W
-				log_message("[W] installed", LOG_MECHA)
-				update_part_values()
-			else
-				to_chat(user, span_warning("There's already a scanning module installed!"))
+	if(is_wire_tool(weapon) && (mecha_flags & PANEL_OPEN))
+		wires.interact(user)
 		return
 
-	if(istype(W, /obj/item/stock_parts/capacitor))
-		if(construction_state == MECHA_OPEN_HATCH)
-			if(!capacitor)
-				if(!user.transferItemToLoc(W, src, silent = FALSE))
-					return
-				to_chat(user, span_notice("You install the capacitor."))
-				playsound(src, 'sound/items/screwdriver2.ogg', 50, FALSE)
-				capacitor = W
-				log_message("[W] installed", LOG_MECHA)
-				update_part_values()
-			else
-				to_chat(user, span_warning("There's already a capacitor installed!"))
+	if(istype(weapon, /obj/item/stock_parts))
+		try_insert_part(weapon, user)
 		return
 
-	if(istype(W, /obj/item/mecha_parts))
-		var/obj/item/mecha_parts/P = W
-		P.try_attach_part(user, src, FALSE)
+	return ..()
+
+/// Try to insert a stock part into the mech
+/obj/vehicle/sealed/mecha/proc/try_insert_part(obj/item/stock_parts/weapon, mob/living/user)
+	if(!(mecha_flags & PANEL_OPEN))
+		balloon_alert(user, "open the panel first!")
 		return
 
-	return ..()
+	if(istype(weapon, /obj/item/stock_parts/cell))
+		if(!cell)
+			if(!user.transferItemToLoc(weapon, src, silent = FALSE))
+				return
+			cell = weapon
+			to_chat(user, span_notice("You install the power cell."))
+			diag_hud_set_mechcell()
+			playsound(src, 'sound/items/screwdriver2.ogg', 50, FALSE)
+			log_message("Power cell installed", LOG_MECHA)
+		else
+			to_chat(user, span_warning("There's already a power cell installed!"))
+		return
+
+	if(istype(weapon, /obj/item/stock_parts/scanning_module))
+		if(!scanmod)
+			if(!user.transferItemToLoc(weapon, src, silent = FALSE))
+				return
+			scanmod = weapon
+			to_chat(user, span_notice("You install the scanning module."))
```

Comment:
Same here wrt balloon alerts. This isn't important information which needs to be persistently logged for the user.